primitives/state-machine/src/overlayed_changes/mod.rs

```diff
@@ -513,13 +561,35 @@ impl OverlayedChanges {
 		root
 	}
 
+	/// Generate the storage root using `backend` and all changes
+	/// as seen by the current transaction.
+	///
+	/// Returns the storage root and do not cache.
+	pub fn storage_root_no_cache<H: Hasher, B: Backend<H>>(
```

Comment:
Why do we need this?

Reply:
I don't remember well, but the subpart of Ext use in cumulus did not include the transaction cache (we do not need transaction content in cumulus).
I remember doing an implementation that do not produce the transaction content and rolling it back (was a bit complex), but an optimisation would be to have trie primitives that do not create the transaction payload.

We could add the 'transaction' cache to 'ExtInner', but only root is really needed for our use case.

Reply:
For now I think it is okay to pass some cache and discard it afterwards.

Reply:
Well it is added ChangeTrieTransaction into the no_std scope, but I can maybe remove the field for no_std.

Reply:
Sounds good